Created ruby-abs-pos-001 test and its reference file

TalbotG opened the pull request:
This is a followup-to [PR29482](https://github.com/web-platform-tests/wpt/pull/29482)

@fantasai wrote over there:

> The abspos test... it's correct, but it's only testing that the abspos isn't trapped by the ruby. To really test the case of abspos within ruby, we'd need to check what happens **when the DIV has 'position: relative' and when the ruby element itself has 'position: relative'**. (But also this last case needs to be made a little clearer in the spec, I think. It should behave as for inlines as specced in https://www.w3.org/TR/CSS2/visudet.html#containing-block-details probably.)
(...)

@fantasai also suggested to use percentage offsets (instead of pixel offsets) like this:
`left: 50%;`
`right: 50%;`
in my old ruby-abs-pos-002.html and ruby-abs-pos-003.html tests over in [PR29482](https://github.com/web-platform-tests/wpt/pull/29482)

After considerable time and efforts, I am confident and happy to come up with this commit.

Over at my website, the test is:

http://www.gtalbot.org/BrowserBugsSection/CSS3Ruby/ruby-abs-pos-004.html

http://www.gtalbot.org/BrowserBugsSection/CSS3Ruby/reference/ruby-abs-pos-004-ref.html

TalbotG committed:
Created ruby-abs-pos-001 test and its reference file

TalbotG committed:
Updated an id attribute

upsuper reviewed (commented):
The comment in https://github.com/web-platform-tests/wpt/pull/35359#pullrequestreview-1064299213 applies here as well.

Also I don't think the reference file is in sync with the test file. For example, in the first subtest, test file has ABCD, but the reference file contains only ABC, and I see no reason D should go away in the test.

TalbotG commented:
In the first subtest, the D is overlapped, masked, covered by the absolutely positioned `<rb id="first-subtest">B</rb>`.

upsuper commented:
Hmm, so the tests are using Ahem font. You should probably add the ahem font stylesheet for them.

Zeuszoid777 commented:
ASent from my Galaxy
-------- Original message --------From: Xidorn Quan ***@***.***> Date: 2022/08/07  02:25  (GMT+02:00) To: web-platform-tests/wpt ***@***.***> Cc: Subscribed ***@***.***> Subject: Re: [web-platform-tests/wpt] Created ruby-abs-pos-001 test and its reference file (PR #35361) 
Hmm, so the tests are using Ahem font. You should probably add the ahem font stylesheet for them.

—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you are subscribed to this thread.Message ID: ***@***.***>

TalbotG commented:
Xidorn,

`<link rel="stylesheet" type="text/css" href="/fonts/ahem.css">`
added now to both
http://www.gtalbot.org/BrowserBugsSection/CSS3Ruby/ruby-abs-pos-004.html

http://www.gtalbot.org/BrowserBugsSection/CSS3Ruby/reference/ruby-abs-pos-004-ref.html

and I created a /font directory and added ahem.css and Ahem.ttf files.

TalbotG commented:
"
If the box has position: absolute:
The containing block is established by the nearest ancestor box that establishes an absolute positioning containing block, in the following way:
(...)
If the ancestor is an inline box,
the containing block is formed by the block-start and inline-start content edges of the first box fragment of the ancestor, and the block-end and inline-end content edges of the last box fragment of the ancestor.
Note: If the ancestor breaks across a line, the 'start' position might more end-ward than the 'end' position.
"
coming from
[CSS Positioned Layout 3, § 2.1. Containing Blocks of Positioned Boxes](https://www.w3.org/TR/css-position-3/#def-cb)
